### ISSUE TITLE
Azure AKS service principal permissions 

### ISSUE THREAD

**Issue by mariojacobo** [kubernetes/autoscaler#1900]
We have updated the CA image version from v1.2.2 to v1.12.3 and are seeing the following issue when trying to scale up/down: 

Normal   ScaledUpGroup         30s   cluster-autoscaler  Scale-up: setting group agentpool size to 2
  Warning  FailedToScaleUpGroup  29s   cluster-autoscaler  Scale-up failed for group agentpool: containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="LinkedAuthorizationFailed" Message="The client '898158bf-REDACTED' with object id '898158bf-REDACTED' has permission to perform action 'Microsoft.ContainerService/managedClusters/write' on scope '/subscriptions/REDACTED/resourceGroups/test-solution-aks-int-rg/providers/Microsoft.ContainerService/managedClusters/test-mag-int-1-aks'; however, it does not have permission to perform action 'Microsoft.OperationalInsights/workspaces/sharedkeys/read' on the linked scope(s) '/subscriptions/REDACTED/resourcegroups/test-solution-infra-int-rg/providers/microsoft.operationalinsights/workspaces/testsolutionintws01'."

The service principal in question is a Log analytics contributor, reader and even subscription owner. I tried all kinds of permutations with these permissions to no avail. What exactly does the service principal need to have ? 

EDIT: with subscription level permissions (contributor), it throws another error: 

Normal   ScaledUpGroup         33s   cluster-autoscaler  Scale-up: setting group agentpool size to 2
  Warning  FailedToScaleUpGroup  32s   cluster-autoscaler  Scale-up failed for group agentpool: containerservice.ManagedClustersClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> <nil>

This is due to the MC* group having a child resource lock (read only) on the NSG. however, child resource s locks should not prevent AKS from scaling up and down. This was working for us for the last year or so, so not entirely sure where and what has changed. 

Note: the NSG lock is required on our end, as scaling up and down has the downside of reverting all NSG rules back to default.

**Comment by bskiba** [comment]
@feiskyer Would you be able to help

**Comment by mariojacobo** [comment]
I actually figured out the issue. The service principal needs to be "log analytics reader" on the workspace resources, not at the resource group level.  So, this change was introduced rather recently, as we had the autoscaler working just fine for the last 6-7 months.  What's even more puzzling is the fact that before we could autoscale while the NSG had a read only lock on it. Now, we have removed it and its working, but not sure why this has changed .
